### PR TITLE
Revert "Revert "Bur 87 frontend create landing page incl homepage""

### PR DESCRIPTION
**Pull request by driesvanspauwen**
Reverts VTKLeuven/burgieclan#100

### CODE REVIEW

**Comment by driesvanspauwen** [comment]
PR 99 was prematurely merged, then reverted in PR 100. This PR reverts that revert, allowing the changes to be reviewed and merged again

File: frontend/src/app/[url_key]/page.tsx [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+    const [page, setPage] = useState<any>(null);
+    const [error, setError] = useState<ApiClientError | null>(null);
+
+    useEffect(() => {
+        const FetchData = async () => {
+            try {
+                const res = await ApiClient('GET', `/api/pages/${url_key}`);
+
+                if (res.error) {
+                    setError({ message: res.error.message, status: res.error.code });
```
Comment (rserry) [inline frontend/src/app/[url_key]/page.tsx:27]: Backend gives inconsistent errors. On 401 it is message and code, on 404 it is detail and status. It should all be detail and status. Ticket is created. This error should be res.error.status and res.error.detail

File: frontend/src/utils/api.tsx [hunk 2]
```diff
@@ -1,27 +1,39 @@
+export type ApiClientError = {
+    message: string;
+    status: string;
```
Comment (rserry) [inline frontend/src/utils/api.tsx:3]: res.error.status is a number

File: frontend/src/utils/api.tsx [hunk 3]
```diff
@@ -1,27 +1,39 @@
...
-    const response = await fetch('/api/proxy', {
-        method: 'POST',
-        headers: {
-            'Content-Type': 'application/json',
-        },
-        body: JSON.stringify({ method, url, body, headers }),
-    });
+    // TODO BUR-110: instead of try-catch, throw errors here and catch with higher-level generic wrapper
+    try {
+        const res = await fetch(frontendBaseUrl + '/api/proxy', {
```
Comment (rserry) [inline frontend/src/utils/api.tsx:21]: Won't /api/proxy be forwarded to backend? Should be /api/frontend/proxy

**Review by rserry — CHANGES_REQUESTED** [review]

**Comment by driesvanspauwen** [comment]
comments will be resolved in https://burgieclan.atlassian.net/browse/BUR-110

File: frontend/src/app/[url_key]/page.tsx (hunk 1 above)
Comment (driesvanspauwen) [reply frontend/src/app/[url_key]/page.tsx:27]: https://burgieclan.atlassian.net/browse/BUR-170

**Pull request merged**